### PR TITLE
fix: prevent possible overflow

### PR DESCRIPTION
**Pull request by armcknight**
The main fix is for a possible overflow when performing a multiplication in some pointer arithmetic.

There are a few unrelated changes rolled in, I'm happy to break them out if desired.

### CODE REVIEW

**Comment by bamx23** [comment]
Hey @armcknight, could you reopen this PR? I'm planning to land some of the latest fixes and make a release.

**Comment by armcknight** [comment]
Sure thing @bamx23!

File: Source/KSCrash/Recording/Tools/KSObjC.c [hunk 1]
```diff
@@ -882,7 +882,7 @@ bool ksobjc_ivarValue(const void* const objectPtr, int ivarIndex, void* dst)
         return false;
     }
     uintptr_t ivarPtr = (uintptr_t)&ivars->first;
-    const struct ivar_t* ivar = (void*)(ivarPtr + ivars->entsizeAndFlags * (unsigned)ivarIndex);
+    const struct ivar_t *ivar = (void *)(ivarPtr + (uintptr_t)ivars->entsizeAndFlags * (uintptr_t)ivarIndex);
```
Comment (bamx23) [inline Source/KSCrash/Recording/Tools/KSObjC.c:885]: May I ask you to preserve project code style?
```suggestion
    const struct ivar_t* ivar = (void*)(ivarPtr + (uintptr_t)ivars->entsizeAndFlags * (uintptr_t)ivarIndex);
```

File: Source/KSCrash/Recording/Monitors/KSCrashMonitor_Signal.c [hunk 2]
```diff
@@ -111,7 +111,6 @@ static void handleSignal(int sigNum, siginfo_t* signalInfo, void* userContext)
     }
 
     KSLOG_DEBUG("Re-raising signal for regular handlers to catch.");
-    // This is technically not allowed, but it works in OSX and iOS.
```
Comment (bamx23) [inline Source/KSCrash/Recording/Monitors/KSCrashMonitor_Signal.c:114]: Why this line was removed?

**Review by bamx23 — CHANGES_REQUESTED** [review]

File: Source/KSCrash/Recording/Monitors/KSCrashMonitor_Signal.c (hunk 2 above)
Comment (bamx23) [reply Source/KSCrash/Recording/Monitors/KSCrashMonitor_Signal.c:114]: Thank for clarification!

**Review by bamx23 — APPROVED** [review]

File: Source/KSCrash/Recording/Monitors/KSCrashMonitor_Signal.c (hunk 2 above)
Comment (armcknight) [reply Source/KSCrash/Recording/Monitors/KSCrashMonitor_Signal.c:114]: My pleasure!

**Pull request merged**